pytket_qir/qir.py

```diff
@@ -141,6 +180,7 @@ def get_required_results(self) -> int:
     def get_optype(self, instr: PyQirInstruction) -> OpType:
         if instr.is_call:
             call_func_name = instr.call_func_name
+            assert call_func_name
```

Comment:
Should this be an exception (or is it logically guaranteed)?

Reply:
It simply is a `mypy` compliance. I can be more explicit as discussed previously. Done.